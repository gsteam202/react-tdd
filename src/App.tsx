import { useState } from 'react';
import '../public/scss/style.scss';
import {
  Container,
  Contents,
  Button,
  InputContainer,
  Input,
  TodoItem,
  TodoListContainer,
} from './Components';

const App = () => {
  const [toDo, setTodo] = useState<string>('');
  const [toDoList, setToDoList] = useState<string[]>([]);

  const handleAdd = () => setToDoList([...toDoList, toDo]);

  const handleDelete = (index: number) => {
    let list = [...toDoList];
    list.splice(index, 1);
    setToDoList(list);
  };

  return (
    <Container>
      <Contents>
        <TodoListContainer data-testid="toDoList">
          {toDoList.map((item, index) => (
            <TodoItem
              key={item}
              label={item}
              onDelete={() => handleDelete(index)}
            />
          ))}
        </TodoListContainer>

        <InputContainer>
          <Input
            placeholder="할 일을 입력해 주세요"
            value={toDo}
            onChange={setTodo}
          />
          <Button
            label="추가"
            backgroundColor="#86efac"
            hoverColor="#0891b2"
            onClick={handleAdd}
          />
        </InputContainer>
      </Contents>
    </Container>
  );
};

export default App;
