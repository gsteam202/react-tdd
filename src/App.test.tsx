import { render, screen } from '@testing-library/react';
import App from './App';

test('render learn react', () => {
  render(<App />);
  const linkEl = screen.getByText(/learn react/i);
  expect(linkEl).toBeInTheDocument();
});
