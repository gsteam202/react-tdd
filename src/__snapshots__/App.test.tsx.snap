// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 렌더링이 잘 되는지 테스트 1`] = `
<div>
  <div
    class="sc-bcXHqe iAIzKk"
  >
    <div
      class="sc-hLBbgP kjVjWS"
    >
      <div
        class="sc-dkrFOg dltswW"
        data-testid="toDoList"
      />
      <div
        class="sc-gswNZR gVBGiX"
      >
        <input
          class="sc-ftTHYK cvWhho"
          placeholder="할 일을 입력해 주세요"
          value=""
        />
        <button
          class="sc-gKPRtg iHulel"
        >
          <div
            class="sc-iBYQkv fXQImq"
          >
            추가
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
